src/server/pfs/cmds/cmds.go

```diff
@@ -877,6 +878,7 @@ $ {{alias}} repo@branch -i http://host/path`,
 	putFile.Flags().StringSliceVarP(&filePaths, "file", "f", []string{"-"}, "The file to be put, it can be a local file or a URL.")
 	putFile.Flags().StringVarP(&inputFile, "input-file", "i", "", "Read filepaths or URLs from a file.  If - is used, paths are read from the standard input.")
 	putFile.Flags().BoolVarP(&recursive, "recursive", "r", false, "Recursively put the files in a directory.")
+	putFile.Flags().BoolVarP(&compress, "compress", "", false, "Compress data during upload.")
```

Comment:
Would there be compat issues with turning this on by default? It seems like you'd always rather use compression.

Reply:
There's two risks in turning it on by default:
- Existing pachds doesn't support GZIP compression, they'll fail with a "No decoder available error". So it would break a 1.11.x client talking to a 1.10.x or earlier pachd. Not sure if that's a concern for us.
- Compression will lead to elevated CPU utilization on the client and server, and additional latency for compressing/decompressing. Depending on the link speed and the compressability of the data it could make performance worse.

I can do some benchmarking to see under what circumstances it's worth enabling compression. I think for now making it available but defaulting it off is less risky.

Reply:
Good reasoning, I agree with keeping it off by default.